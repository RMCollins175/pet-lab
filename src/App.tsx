import "./styles.css";
import React, { useEffect } from "react";
import { ProductsList } from "./components/ProductsList/ProductsList";
import { Filters } from "./components/Filters/Filters";
import { useFetchProducts } from "./components/utilities/hooks/useFetchProducts";

function App() {
  const { fetchProducts, error, handleSearch, isLoading, products } =
    useFetchProducts();

  useEffect(() => {
    fetchProducts();
  }, []);

  if (error) {
    return <h1>Sorry, please try again later</h1>;
  }

  return (
    <main className="container">
      <Filters handleSearch={handleSearch} loading={isLoading} />
      <ProductsList products={products} />
    </main>
  );
}

export default App;
