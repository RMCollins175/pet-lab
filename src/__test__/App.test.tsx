import { mount } from "enzyme";
import App from "../App";

const elem = <App />;
const wrapper = mount(elem);

describe("App", () => {
  it("should render an App component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should fetch products from the API when the component is mounted", () => {});
});
