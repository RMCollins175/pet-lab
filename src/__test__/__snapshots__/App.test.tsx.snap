// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render a default App component 1`] = `ReactWrapper {}`;

exports[`App should render an App component 1`] = `ReactWrapper {}`;

exports[`ProductsList component tests should render a default productList component 1`] = `ReactWrapper {}`;
